lib/grandpa/grandpa_test.go

```diff
@@ -64,30 +67,34 @@ func newTestState(t *testing.T) *state.Service {
 	block, err := state.NewBlockStateFromGenesis(db, testGenesisHeader)
 	require.NoError(t, err)
 
+	grandpa, err := state.NewGrandpaStateFromGenesis(db, voters)
+	require.NoError(t, err)
+
 	return &state.Service{
-		Block: block,
+		Block:   block,
+		Grandpa: grandpa,
 	}
 }
 
 func newTestVoters() []*Voter {
-	voters := []*Voter{}
+	vs := []*Voter{}
```

Comment:
Replace it with a nil slice.
```
var vs []*Voter
```

Reply:
is there a practical difference between the two? I thought `var vs []*Voter` would make `[]*Voter(nil)` and `voters := []*Voter{}` makes `[]*Voter{}`, so one is nil and one is empty, but maybe they are the same?